tests/schema/tests.py

```diff
@@ -1749,6 +1749,20 @@ def test_indexes(self):
             self.get_indexes(Book._meta.db_table),
         )
 
+    @skipUnlessDBFeature('supports_index_on_text_field')
+    def test_text_field_with_db_index(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(AuthorTextFieldWithIndex)
+        # The text_field index should be present
+        self.assertIn('text_field', self.get_indexes(AuthorTextFieldWithIndex._meta.db_table))
```

Comment:
Perhaps this would be better as a single test:
```
assertion = self.assertIn if connection.features.supports_index_on_text_field else self.assertNotIn
assertion('text_field', self.get_indexes(AuthorTextFieldWithIndex._meta.db_table))
```